File: Problems/0705. (E) Design Hashset/Java/solution.java
class MyHashSet {
  // store all elements in 2D list

  private List<Integer>[] items;
  private boolean max;

  public MyHashSet() {
    this.items = new List[1000];
    for (int i = 0; i < 1000; i++)
      this.items[i] = new ArrayList();
    this.max = false;
  }
  
  public void add(int key) {
    if (key == 1000000)
      this.max = true;
    else {
      int mod = key % 1000;
      int quot = key / 1000;
      if (!this.contains(key))
        this.items[quot].add(mod);
    }
  }
  
  public void remove(int key) {
    if (key == 1000000)
      this.max = false;
    else {
      int mod = key % 1000;
      int quot = key / 1000;
      this.items[quot].remove(new Integer(mod));
    }
  }
  
  public boolean contains(int key) {
    if (key == 1000000)
      return this.max;
    int mod = key % 1000;
    int quot = key / 1000;
    return this.items[quot].contains(mod);
  }
}

File: Problems/0705. (E) Design Hashset/Python3/solution.py
class MyHashSet:
  # store all elements in 2D list
  
  def __init__(self):
    self._items = [[] for _ in range(1000)]
    self._max = False

  def add(self, key: int) -> None:
    if key == 1000000:
      self._max = True
    else:
      mod = key % 1000
      quot = key // 1000
      if not self.contains(key):
        self._items[quot].append(mod)

  def remove(self, key: int) -> None:
    if key == 1000000:
      self._max = False
    else:
      mod = key % 1000
      quot = key // 1000
      try:
        self._items[quot].remove(mod)
      except:
        pass
      
  def contains(self, key: int) -> bool:
    if key == 1000000:
      return self._max
    mod = key % 1000
    quot = key // 1000
    return True if mod in self._items[quot] else False

